src/Avalonia.Base/Input/FocusManager.cs

```diff
@@ -245,5 +279,887 @@ private static bool IsVisible(IInputElement e)
+        {
+            var root = (_contentRoot as Visual)?.GetSelfAndVisualDescendants().FirstOrDefault(x => x is IInputElement) as IInputElement;
+            if (root == null)
+                return null;
+            return GetFirstFocusableElementFromRoot(false);
+        }
+
+        /// <summary>
+        /// Retrieves the first element that can receive focus based on the specified scope.
+        /// </summary>
+        /// <param name="searchScope">The root element from which to search.</param>
+        /// <returns>The first focusable element.</returns>
+        public static IInputElement? FindFirstFocusableElement(IInputElement searchScope)
+        {
+            return GetFirstFocusableElement(searchScope);
+        }
+
+        /// <summary>
+        /// Retrieves the last element that can receive focus.
+        /// </summary>
+        /// <returns>The last focusable element.</returns>
+        public IInputElement? FindLastFocusableElement()
+        {
+            var root = (_contentRoot as Visual)?.GetSelfAndVisualDescendants().FirstOrDefault(x => x is IInputElement) as IInputElement;
+            if (root == null)
+                return null;
+            return GetFirstFocusableElementFromRoot(true);
+        }
+
+        /// <summary>
+        /// Retrieves the last element that can receive focus based on the specified scope.
+        /// </summary>
+        /// <param name="searchScope">The root element from which to search.</param>
+        /// <returns>The last focusable object.</returns>
+        public static IInputElement? FindLastFocusableElement(IInputElement searchScope)
+        {
+            return GetFocusManager(searchScope)?.GetLastFocusableElement(searchScope);
+        }
+
+        /// <summary>
+        /// Retrieves the element that should receive focus based on the specified navigation direction.
+        /// </summary>
+        /// <param name="direction"></param>
+        /// <returns></returns>
+        public IInputElement? FindNextElement(NavigationDirection direction)
+        {
+            var xyOption = new XYFocusOptions()
+            {
+                UpdateManifold = false
+            };
+
+            return FindNextFocus(direction, xyOption);
+        }
+
+        /// <summary>
+        /// Retrieves the element that should receive focus based on the specified navigation direction (cannot be used with tab navigation).
+        /// </summary>
+        /// <param name="direction">The direction that focus moves from element to element within the app UI.</param>
+        /// <param name="options">The options to help identify the next element to receive focus with the provided navigation.</param>
+        /// <returns>The next element to receive focus.</returns>
+        public IInputElement? FindNextElement(NavigationDirection direction, FindNextElementOptions options)
+        {
+            if (direction is not NavigationDirection.Up
```

Comment:
The directional navigation validation can also be moved to `CreateFocusOptions` (or to its own method) and shared between `FindNextElement` and `MoveFocus`